pkg/api/grpcServer.go

```diff
@@ -155,14 +169,22 @@ func (s *GRPCServer) streamLoop(stream grpc.ServerStream, request proto.Message,
 			return err
 		}
 
+		var span ot.Span
```

Comment:
Opentracing is written to default to a no-op tracer, so the call to StartSpanFromContext amounts to [extracting a key from the context](https://github.com/opentracing/opentracing-go/blob/master/gocontext.go#L43) and [returning a static value (the no-op span)](https://github.com/opentracing/opentracing-go/blob/master/noop.go#L52). This shouldn't even result in an allocation, other than the span we declare here (and we could even hoist this declaration out of the loop so the allocation only occurs at stream creation, though I'd expect the compiler to perform that optimization already). [Calls on the no-op span themselves do no work](https://github.com/opentracing/opentracing-go/blob/master/noop.go#L37-L49).

I thought that was sufficiently low overhead that keeping the code simpler by not introducing branches was worth it. A different option we have is to keep the is-tracing-enabled state in some object that's a wrapper around the tracer, and use that object to create spans. It could short-circuit to the no-op span if tracing isn't enabled, while still keeping this code conceptually condition free.